test/math_test.exs

```diff
@@ -35,4 +35,90 @@ defmodule Quantity.MathTest do
       assert_raise(ArgumentError, fn -> Quantity.sum!([~Q[123 bananas], ~Q[456 apples]], 0, "apples") end)
     end
   end
+
+  describe "inverse" do
+    test "inverse of a quantity with simple unit" do
+      assert Quantity.inverse(~Q[10 banana]) == ~Q[0.1 1/banana]
+    end
+
+    test "inverse of a quantity with 1/unit" do
+      assert Quantity.inverse(~Q[0.1 1/banana]) == ~Q[1E1 banana]
+    end
+
+    test "inverse of a quantity with div unit" do
+      assert Quantity.inverse(~Q[10 banana/hour]) == ~Q[0.1 hour/banana]
+    end
+
+    test "using 1/unit quantities in other Math functions" do
+      assert ~Q[10 banana]
+             |> Quantity.inverse()
+             |> Quantity.mult(~d[50])
+             |> Quantity.add!(~Q[5 1/banana])
+             |> Quantity.inverse() ==
+               ~Q[0.1 banana]
+    end
+  end
+
+  describe "div" do
+    test "dividing by quantity without unit" do
+      assert Quantity.div(~Q[25 bananas], ~Q[5]) == ~Q[5 bananas]
+    end
+
+    test "dividing by 1/unit" do
+      assert Quantity.div(~Q[25 bananas], ~Q[5 1/s]) == ~Q[5 bananas*s]
+    end
+
+    test "dividing 1 unit with regular quantity" do
+      assert Quantity.div(~Q[25], ~Q[5 bananas]) == ~Q[5 1/bananas]
+    end
+  end
+
+  describe "mult" do
+    test "multiplying by quantity without unit" do
+      assert Quantity.mult(~Q[12 bananas], ~Q[3]) == ~Q[36 bananas]
+    end
+
+    test "multiplying by quantity with 1/unit" do
+      assert Quantity.mult(~Q[12 bananas], ~Q[3 1/s]) == ~Q[36 bananas/s]
+    end
+
+    test "multiplying by quantity with 1/unit that matches" do
+      assert Quantity.mult(~Q[12 s], ~Q[3 1/s]) == ~Q[36]
+      assert Quantity.mult(~Q[12 s*s], ~Q[3 1/s]) == ~Q[36 s]
+      assert Quantity.mult(~Q[12 s/banana], ~Q[3 1/s]) == ~Q[36 1/banana]
+    end
+
+    test "multiplying with inverse unit" do
+      assert Quantity.mult(~Q[12 s/banana], ~Q[3 banana/s]) == ~Q[36]
+    end
+
+    test "when unit does not produce anything valid, it still returns something" do
```

Comment:
The title here states "anything valid", but now it is actually valid (outputting `~Q[34 apples*bananas/s*s]`).. Perhaps rename to "multiplies a complex unit" or something similar?

Reply:
Ah yes, forgot to fix that test. Done in 69e2415

Reply:
Nice :+1: